src/main/java/com/example/controllers/CurrencyController.java

```diff
@@ -0,0 +1,118 @@
+package com.example.controllers;
+
+import java.io.IOException;
+import java.util.List;
+
+import javax.annotation.PostConstruct;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.scheduling.annotation.EnableScheduling;
+import org.springframework.scheduling.annotation.Scheduled;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+import com.example.repository.CurrencyRepository;
+
+import model.Currency;
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.Response;
+
+@RestController
+@EnableScheduling
+public class CurrencyController {
+
+	private static final int START_OF_CURRENCY = 3;
+	private static final int CURRENCY_NAME_LINE = 4;
+	private static final int CURRENCY_SYMBOL_LINE = 8;
+	private static final int CURRENCY_USDPRICE_LINE = 20;
+	private static final int CURRENCY_PERCENTCHANGE_LINE = 22;
+	private static final int END_OF_CURRENCY = 30;
+	private static final int LAST_LINE = 300;
+	
+	@Autowired
+	CurrencyRepository cr;
+
+	@RequestMapping("/putCurrencies")
+	@PostConstruct
+	@Scheduled(cron = "0 * * ? * *")
+	public String putCurrencies() {
+
+		OkHttpClient client = new OkHttpClient().newBuilder().build();
```

Comment:
Don't add business logic to the controller directly. Create additional layer (service) and add code there.